drivers/net/wireless/mediatek/mt76/mt76_connac_mcu.c

```diff
@@ -1552,6 +1552,21 @@ int mt76_connac_mcu_set_deep_sleep(struct mt76_dev *dev, bool enable)
 }
 EXPORT_SYMBOL_GPL(mt76_connac_mcu_set_deep_sleep);
 
+int mt76_connac_sta_state_dp(struct mt76_dev *dev,
+			     enum ieee80211_sta_state old_state,
+			     enum ieee80211_sta_state new_state) {
+	bool dp = false;
+
+	if (new_state == IEEE80211_STA_AUTHORIZED ||
+	    new_state == IEEE80211_STA_NOTEXIST)
+		dp = true;
+
+	mt76_connac_mcu_set_deep_sleep(dev, dp);
+
+	return 0;
+}
+EXPORT_SYMBOL_GPL(mt76_connac_sta_state_dp);
+
 void mt76_connac_mcu_coredump_event(struct mt76_dev *dev, struct sk_buff *skb,
```

Comment:
@objelf in order to reduce unnecessary mcu cmds we can do something like:
enable deep sleep switching from IEEE80211_STA_NONE to IEEE80211_STA_NOTEXIST and from IEEE80211_STA_ASSOC to IEEE80211_STA_AUTHORIZED while we can disable it switching from IEEE80211_STA_NOTEXIST to IEEE80211_STA_NONE, what do you think?

Reply:
ok, i will try this way and do some stress test.